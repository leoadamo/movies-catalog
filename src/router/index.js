// DEPENDENCIES
import { createRouter, createWebHistory } from "vue-router";

const routes = [
  {
    path: "/",
    name: "home",
    component: () => import("@/views/HomePage.vue"),
  },
  {
    path: "/movies/:id/:slug",
    name: "movie-detail",
    component: () => import("@/views/MovieDetail.vue"),
    props: true,
    beforeEnter: async (to, from) => {
      const store = await import("@/stores/movies");

      const { getMovieById } = store.useMoviesStore();

      const exists = getMovieById(to.params.id);

      if (!exists) {
        return {
          name: "not-found",
          params: { pathMatch: to.path.split("/").slice(1) },
          query: to.query,
          hash: to.hash,
        };
      }
    },
  },
  {
    path: "/:pathMatch(.*)*",
    name: "not-found",
    component: () => import("@/views/NotFound.vue"),
  },
];

const router = createRouter({
  history: createWebHistory(),
  routes,
});

export default router;
