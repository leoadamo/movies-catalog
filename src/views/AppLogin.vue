<script setup>
// DEPENDENCIES
import { ref, useTemplateRef, onMounted } from "vue";

// COMPONENTS
import { ChevronRightIcon } from "@heroicons/vue/24/solid";

// REFS
const form = ref({
  username: "",
  password: "",
});

const userNameInputRef = useTemplateRef("username-input");

// METHODS
function handleLogin() {
  console.log("Logging in with:", form.value);

  // Reset form after login attempt
  form.value.username = "";
  form.value.password = "";
}

// LIFECYCLE HOOKS
onMounted(() => {
  userNameInputRef.value?.focus();
});
</script>

<template>
  <div
    class="w-full max-w-[400px] m-auto grid grid-cols-1 gap-8 p-8 text-white bg-gray-800 rounded-lg shadow-lg"
  >
    <div class="flex flex-col gap-1">
      <h1 class="text-2xl font-bold">Log In</h1>
      <p>Please, inform your credentials to continue.</p>
    </div>

    <form class="grid grid-cols-1 gap-8" @submit.prevent="handleLogin">
      <div class="flex flex-col gap-1">
        <label for="username">Username:</label>
        <input
          v-model="form.username"
          ref="username-input"
          id="username"
          type="text"
          class="border rounded border-gray-400 bg-gray-900"
          autocomplete="off"
          required
        />
      </div>

      <div class="flex flex-col gap-1">
        <label for="password">Password:</label>
        <input
          v-model="form.password"
          id="password"
          type="password"
          class="border rounded border-gray-400 bg-gray-900"
          autocomplete="off"
          required
        />
      </div>

      <button type="submit" class="app-button app-button--primary ml-auto">
        Continue

        <chevron-right-icon class="w-5 h-5" aria-hidden="true" />
      </button>
    </form>
  </div>
</template>
