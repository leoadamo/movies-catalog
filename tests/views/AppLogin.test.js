// DEPENDENCIES
import { describe, it, expect, beforeEach } from "vitest";
import { render } from "@testing-library/vue";
import { createTestingPinia } from "@pinia/testing";
import { createRouter, createWebHistory } from "vue-router";

// ROUTES
import { routes } from "@/router";

// COMPONENTS
import AppLogin from "@/views/AppLogin.vue";

let router;
let pinia;

beforeEach(() => {
  router = createRouter({
    history: createWebHistory(),
    routes,
  });

  pinia = createTestingPinia();
});

describe("App Login", () => {
  it("should render the login view", () => {
    const wrapper = render(AppLogin, {
      global: {
        plugins: [router, pinia],
      },
    });

    const login = wrapper.queryByTestId("app-login");

    expect(login).toBeTruthy();
    expect(login).toBeInTheDocument();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
