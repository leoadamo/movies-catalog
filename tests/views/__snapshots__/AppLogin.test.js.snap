// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App Login > should render the login view 1`] = `
"<div class="w-full max-w-[400px] m-auto grid grid-cols-1 gap-8 p-8 text-white bg-gray-800 rounded-lg shadow-lg" data-testid="app-login">
  <div class="flex flex-col gap-1">
    <h1 class="text-2xl font-bold">Log In</h1>
    <p>Please, inform your credentials to continue.</p>
  </div>
  <form class="grid grid-cols-1 gap-5">
    <div class="relative grid grid-cols-1 gap-8 pb-5">
      <div class="flex flex-col gap-1"><label for="username">Username:</label><input id="username" type="text" data-testid="username-input" class="border rounded border-gray-400 bg-gray-900" autocomplete="off" required=""></div>
      <div class="flex flex-col gap-1"><label for="password">Password:</label><input id="password" type="password" class="border rounded border-gray-400 bg-gray-900" autocomplete="off" required=""></div>
      <transition-stub name="fade" appear="false" persisted="false" css="true">
        <!--v-if-->
      </transition-stub>
    </div><button type="submit" class="app-button app-button--primary ml-auto"> Continue <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="currentColor" aria-hidden="true" data-slot="icon" class="w-5 h-5">
        <path fill-rule="evenodd" d="M16.28 11.47a.75.75 0 0 1 0 1.06l-7.5 7.5a.75.75 0 0 1-1.06-1.06L14.69 12 7.72 5.03a.75.75 0 0 1 1.06-1.06l7.5 7.5Z" clip-rule="evenodd"></path>
      </svg></button>
  </form>
</div>"
`;
